### DISCUSSION TITLE
Diagnostic plots for detecting autoregressive and non-stationary noise

### DISCUSSION THREAD

**Discussion by rccreswell** [pints-team/pints#1670] [Ideas/Science]
@ben18785 and I have been looking at some additional diagnostic plots for residuals. These are helpful for detecting when an IID noise process is inadequate and correlation or non-stationary behaviour is present. Functionality for these plots could be added to Pints in the pints.residuals_diagnostics module.

Three diagnostics which could be added are:

1. The residuals are divided into bins, and the lag 1 autocorrelation within each bin is calculated. The binned autocorrelations are plotted over time.
2. Same as above, but looking at the variance within each bin. 
3. A distance matrix between the residuals. When a banded structure appears in the distance matrix, this suggests that the true noise process may be autocorrelated.

**Comment by MichaelClerx** [comment]
Are these new ideas, or are they commonly used plots?

I think this is great work, but a bit worried how we're going to write documentation etc. without papers to refer to for details.

**Comment by ben18785** [comment]
@MichaelClerx these will be in Richard's paper that will be submitted soon. I don't think we should be too led by publications for what to include / not though; otherwise, the whole of Pints wouldn't really be in there ;)

**Comment by MichaelClerx** [comment]
Haha, fair enough! As long as the end result makes sense to outsiders I'm happy. (I guess the notebooks take care of that!)
